docs/architecture/ingest-event-path.md

```diff
@@ -0,0 +1,151 @@
+
+*  Rate limits are not calculated using Redis, instead Relay just honors 429s
+   from previously mentioned endpoint.
+
+*  Filters are not applied at all.
+
+## Inside the endpoint
+
+When an SDK hits `/api/X/store` on Relay, the code in
+`server/src/endpoints/store.rs` is called before returning a HTTP response.
+
+That code looks into an in-memory cache to answer basic information about a project such as:
+
+*  Does it exist? Is it suspended/disabled?
+
+*  Is it rate limited right now? If so, which key is rate limited?
+
+*  Which DSNs are valid for this project?
+
+Some of the data for this cache is coming from the [projectconfigs
+endpoint](https://github.com/getsentry/sentry/blob/c868def30e013177383f8ca5909090c8bdbd8f6f/src/sentry/api/endpoints/relay_projectconfigs.py).
+It is refreshed every couple of minutes, depending on configuration.
+
+If the cache is fresh, we may return a `429` for rate limits or a `4xx` for
+invalid auth information.
+
+That cache might be empty or stale. If that is the case, Relay does not
+actually attempt to populate it at this stage. **It just returns a `200` even
+though the event might be dropped later.** This implies:
+
+*  The first store request that runs into a rate limit doesn't actually result
+   in a `429`, but a subsequent request will (because by that time the project
+   cache will have been updated).
+
+*  A store request for a non-existent project may result in a `200`, but
+   subsequent ones will not.
+
+*  A store request with wrong auth information may result in a `200`, but
+   subsequent ones will not.
+
+*  Filters are also not applied at this stage, so **a filtered event will
+   always result in a `200`**. This matches the Python behavior since [a while
+   now](https://github.com/getsentry/sentry/pull/14561).
+
+These examples assume that a project receives one event at a time. In practice
+one may observe that a highly concurrent burst of store requests for a single
+project results in `200 OK`s only. However, a multi-second flood of incoming
+events should quickly result in eventually consistent and correct status codes.
+
+The response is completed at this point. All expensive work (such as talking to
+external services) is deferred into a background task. Except for responding to
+the HTTP request, there's no I/O done in the endpoint in any form. We didn't
+even hit Redis to calculate rate limits.
+
+The fast response times are supposed to benefit application environments which
+cannot send HTTP requests fully asynchronously (PHP and certain serverless
+platorms), at the cost of status codes being inaccurate. Note that we still
+emit accurate outcomes into Kafka if configured to do so.
+
+## The background task
+
+The HTTP response is out, with a status code that is just a best-effort guess
+at what the actual outcome of the event is going to be. The rest of what used
+to happen synchronously in the Python `StoreView` is done asynchronously, but
+still in the same process.
```

Comment:
Maybe slightly more clear would be: 

The HTTP response returned is just a best-effort guess at what the actual outcome of the event is going to be. We only return a 4XX code if we know that the response will fail (based on cached information), if we don't we return a 200 and continue to process the event asynchronously. This asynchronous  processing used to happen synchronously in the Python implementation of `StoreView`.  

The effect of this is that the server will respond much faster that before but we might return 200 for events that will ultimately not be accepted.